upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -95,7 +97,50 @@ func (b *BootstrapChannelBuilder) Build(c *fi.ModelBuilderContext) error {
 			Location:  fi.String(manifestPath),
 			Name:      fi.String(name),
 		}
+	}
+
+	if b.ClusterAddons != nil {
+		key := "cluster-addons.kops.k8s.io"
+		version := "0.0.0"
+		location := key + "/default.yaml"
+
+		a := &channelsapi.AddonSpec{
+			Name:     fi.String(key),
+			Version:  fi.String(version),
+			Selector: map[string]string{"k8s-addon": key},
+			Manifest: fi.String(location),
+		}
+
+		if a.Id != "" {
+			key = key + "-" + a.Id
+		}
+		name := b.Cluster.ObjectMeta.Name + "-addons-" + key
+		manifestPath := "addons/" + *a.Manifest
+
+		manifestBytes, err := b.ClusterAddons.ToYAML()
+		if err != nil {
+			return fmt.Errorf("error serializing addons: %v", err)
+		}
+
+		// Trim whitespace
+		manifestBytes = []byte(strings.TrimSpace(string(manifestBytes)))
+
+		rawManifest := string(manifestBytes)
+
+		manifestHash, err := utils.HashString(rawManifest)
+		if err != nil {
+			return fmt.Errorf("error hashing manifest: %v", err)
+		}
+		a.ManifestHash = manifestHash
+
+		tasks[name] = &fitasks.ManagedFile{
```

Comment:
Why not call `c.AddTask()`? This seems an indirect, hacky way of adding a task.

Reply:
Agreed, and I don't know why.  Changed to use `AddTask`